interpreter/src/main/scala/pl/touk/nussknacker/engine/definition/component/ComponentStaticDefinition.scala

```diff
@@ -1,19 +1,36 @@
 package pl.touk.nussknacker.engine.definition.component
 
+import pl.touk.nussknacker.engine.api.component.ComponentType.ComponentType
 import pl.touk.nussknacker.engine.api.component.SingleComponentConfig
 import pl.touk.nussknacker.engine.api.definition.Parameter
 import pl.touk.nussknacker.engine.api.typed.typing.TypingResult
 
 case class ComponentStaticDefinition(
+    override val componentType: ComponentType,
     parameters: List[Parameter],
     returnType: Option[TypingResult],
     categories: Option[List[String]],
-    componentConfig: SingleComponentConfig
-) {
+    componentConfig: SingleComponentConfig,
+    componentTypeSpecificData: ComponentTypeSpecificData
+) extends BaseComponentDefinition {
 
   def withComponentConfig(componentConfig: SingleComponentConfig): ComponentStaticDefinition =
     copy(componentConfig = componentConfig)
 
   val hasNoReturn: Boolean = returnType.isEmpty
 
 }
+
+sealed trait ComponentTypeSpecificData
```

Comment:
Can't this be an Option[CustomComponentSpecificData]?

Reply:
I'm not sure about that. I was considering even adding additional `*ComponentSpecificData` which will extends `NoComponentTypeSpecificData` and have some assertion that each `ComponentType` has related `ComponentSpecificData`. Thanks to that we will be sure that if we check `componentType` before, e.g. `asCustomComponentData` won't throw an `Exception`. `Option[ComponentTypeSpecificData]` could be interpreted by someone as a data that we don't know yet and I intended it to be a kind of inheritance but used by composition. Is it worth it? WDYT?